### PR TITLE
NUTCH-1741 sitemaps argument for generate, fetch, and parse, takes te…

### PR DESCRIPTION
**Pull request by benmvachon**
…rnary operator ('none', 'only', 'all')

### CODE REVIEW

File: src/java/org/apache/nutch/crawl/SitemapOperation.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.apache.nutch.crawl;
```
Comment (lewismc) [inline src/java/org/apache/nutch/crawl/SitemapOperation.java]: Please add license header

File: src/java/org/apache/nutch/crawl/SitemapOperation.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+public enum SitemapOperation {
+  NONE,
+  ONLY,
+  ALL;
+  
+  public static SitemapOperation get(String operation) {
```
Comment (lewismc) [inline src/java/org/apache/nutch/crawl/SitemapOperation.java]: Document method

File: src/plugin/parse-sitemap/src/test/org/apache/nutch/parse/sitemap/TestNutchSitemapParser.java [hunk 3]
```diff
@@ -15,40 +15,30 @@
...
   public void setUp() throws Exception {
-    super.setUp();
-    page =mock(WebPage.class);
-  }
-
-  @Override
-  @After
-  public void tearDown() throws Exception {
-    super.tearDown();
+    page = new WebPage();
```
Comment (lewismc) [inline src/plugin/parse-sitemap/src/test/org/apache/nutch/parse/sitemap/TestNutchSitemapParser.java]: I think this should use the full constructor
```
WebPage.newBuilder().build();;
```

File: src/plugin/parse-sitemap/src/java/org/apache/nutch/parse/sitemap/NutchSitemapParser.java [hunk 4]
```diff
@@ -62,7 +68,6 @@ public NutchSitemapParse getParse(String url, WebPage page) {
       LOG.error(StringUtils.stringifyException(e));
     }
     Map<Outlink, Metadata> outlinkMap = null;
-    Iterator i$;
     if (siteMap.isIndex()) {
```
Comment (benmvachon) [inline src/plugin/parse-sitemap/src/java/org/apache/nutch/parse/sitemap/NutchSitemapParser.java]: Can someone walk me through how sitemap indexes are supposed to be being handled?
It doesn't seem to me like sitemaps in the WebPage.sitemaps field are ever actually added to the crawler store. And if they are, they are not marked with the sitemap inject marker.

File: src/java/org/apache/nutch/parse/ParseUtil.java [hunk 5]
```diff
@@ -261,19 +263,14 @@ public void processSitemapParse(String url, WebPage page,
 
           Mark.INJECT_MARK.putMark(newRow, InjectType.SITEMAP_INJECT.getTypeString());
```
Comment (benmvachon) [inline src/java/org/apache/nutch/parse/ParseUtil.java]: Isn't the sitemap inject marker supposed to make pages AS sitemaps? Why is it being applied to links that were found in sitemaps?

File: src/plugin/parse-sitemap/src/java/org/apache/nutch/parse/sitemap/NutchSitemapParser.java (hunk 4 above)
Comment (benmvachon) [reply src/plugin/parse-sitemap/src/java/org/apache/nutch/parse/sitemap/NutchSitemapParser.java]: Here too: https://github.com/apache/nutch/pull/304/files#diff-44035662a89a7d5fb73289b33bc0dab9R539
I don't see where this is actually adding the sitemap to the datastore for generate jobs.

Comment (benmvachon) [reply src/plugin/parse-sitemap/src/java/org/apache/nutch/parse/sitemap/NutchSitemapParser.java]: The only other place the WebPage.sitemaps field is used is here: https://github.com/benmvachon/nutch/blob/f314baf3d346f58c407977b65283fcf3e23055b5/src/java/org/apache/nutch/crawl/DbUpdateMapper.java#L72

**Comment by abratnap** [comment]
Seems this PR is still open and not merged.. Though it's mentioned in 2.4 release notes that the sitemap generation is supported.
https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10680&version=12324540

**Comment by sebastian-nagel** [comment]
NUTCH-1741 has been implemented in ffa04e1. This PR adds the option to control what is fetched in a batch - only sitemaps, only "ordinary" web pages, or both.

We had to retire the development on the 2.x branch after the release of 2.4 as no committers are actively working on it. Sorry about this. But thanks for your contribution, @benmvachon! I'll go through PRs and close the ones for 2.x which have been left open.

**Comment by abratnap** [comment]
Thank you @sebastian-nagel.

**Pull request closed without merging**